src/main.rs

```diff
@@ -59,48 +64,92 @@ async fn main() {
-                match get::get_table(config.init().await.unwrap(), table, shape, format).await {
-                    Ok(x) => x,
-                    Err(x) => format!("ERROR: {:?}", x),
-                };
-            Ok(result)
-        }
-        Some(("serve", _sub_matches)) => serve::app(config.init().await.unwrap()),
-        _ => unreachable!("Exhausted list of subcommands and subcommand_required prevents `None`"),
-    };
+                        init::init(&config).await
+                    }
+                    _ => init::init(&config).await,
+                },
+                Some(("config", _sub_matches)) => Ok(config.to_string()),
+                Some(("get", sub_matches)) => {
+                    let table = match sub_matches.get_one::<String>("TABLE") {
+                        Some(x) => x,
+                        _ => panic!("No table given"),
+                    };
+                    let shape = match sub_matches.get_one::<String>("shape") {
+                        Some(x) => x,
+                        _ => "value_rows",
+                    };
+                    let format = match sub_matches.get_one::<String>("format") {
+                        Some(x) => x,
+                        _ => "text",
+                    };
+                    let result =
+                        match get::get_table(config.init().await.unwrap(), table, shape, format)
+                            .await
+                        {
+                            Ok(x) => x,
+                            Err(x) => format!("ERROR: {:?}", x),
+                        };
+                    Ok(result)
+                }
+                Some(("serve", _sub_matches)) => serve::app(config.init().await.unwrap()),
+                _ => unreachable!(
+                    "Exhausted list of subcommands and subcommand_required prevents `None`"
+                ),
+            },
+        };
 
     //print exit message
     match exit_result {
         Err(x) => {
-            println!("{}", x);
+            tracing::error!("{}", x);
             std::process::exit(1)
         }
 
         Ok(x) => println!("{}", x),
     }
 }
+
+#[tokio::main]
+async fn handle_cgi(vars: &HashMap<String, String>, config: &mut Config) -> Result<String, String> {
+    let shared_state = Arc::new(serve::AppState {
+        config: config.init().await.unwrap().clone(),
+    });
+    let app = build_app(shared_state);
+
+    let client = TestClient::new(app);
+    // TODO: Replace this dummy request with a request that is build from `vars`:
+    let res = client.get("/table").send().await;
+
+    Ok(String::from("CGI request handled successfully!"))
+}
+
+fn cgi_vars() -> Option<HashMap<String, String>> {
+    let mut vars = match env::var_os("GATEWAY_INTERFACE").and_then(|p| Some(p.into_string())) {
+        Some(Ok(s)) if s == "CGI/1.1" => HashMap::new(),
+        _ => return None,
+    };
+
+    for var in vec!["REQUEST_METHOD", "PATH_INFO", "QUERY_STRING"] {
```

Comment:
I think we can have sensible defaults for these instead of failing:

- REQUEST_METHOD should default to GET
- PATH_INFO should default to `/` but maybe `/table` for now?
- QUERY_STRING should default to the empty string, I think